### PR TITLE
NSH: first implementation and small update

### PR DESCRIPTION
**Pull request by GabrielGanne**
Hi,

I took an implementation of NSH I found by @christophefontaine last year, upon which I added a small cleaning and protocol update.
(I found his commits here: https://bitbucket.org/secdev/scapy/pull-requests/107/vxlan-nsh/diff)

I'll be leaving on holidays, and I'm not planing on having much of an internet access ; I'll get back to this next month, and answer or fix your reviews.

Regards

### CODE REVIEW

File: scapy/contrib/vxlan.py [hunk 1]
```diff
@@ -1,26 +1,40 @@
+#! /usr/bin/env python
 # RFC 7348 - Virtual eXtensible Local Area Network (VXLAN):
 # A Framework for Overlaying Virtualized Layer 2 Networks over Layer 3 Networks
 # http://tools.ietf.org/html/rfc7348
-
-# scapy.contrib.description = VXLAN
+# http://datatracker.ietf.org/doc/draft-quinn-vxlan-gpe/
```
Comment (guedou) [inline scapy/contrib/vxlan.py]: This link does not seem uptodate. Is https://datatracker.ietf.org/doc/draft-ietf-nvo3-vxlan-gpe/ the correct one ?

**Comment by guedou** [comment]
Thanks for this PR ! Would you mind adding a .uts file and some simple unit tests ? It would ease the usage of this module by providing examples.

Also, could you clean up the commits (for example using  "git rebase -i" ) ? A single commit is much easier for Scapy history.

File: scapy/contrib/vxlan.py (hunk 1 above)
Comment (GabrielGanne) [inline scapy/contrib/vxlan.py]: You're right. I forgot to update it.

**Comment by GabrielGanne** [comment]
Yes, I will add unit tests. (Actually, I forgot about those and it seems I am breaking VXLAN's ... I'll fix them on the way).

Rebasing is not an issue. Do you wish for only one commit ? or one for each layer (VXLAN, and NSH) ?

**Comment by guedou** [comment]
A commit per file will be better. Yet, a single commit is also OK as the history is also clean.

**Comment by GabrielGanne** [comment]
Hi,

It seems I'm breaking travis' tests, but I'm unable to reproduce this issue on the computer I currently have (probably some env I'm not aware about).

I'll try and fix this when I'll be back on my computer.

Regards.

**Comment by GabrielGanne** [comment]
It should now work as expected.

**Comment by GabrielGanne** [comment]
Hi,
Sorry I missed that. I'll look how to fix this.

**Pull request merged**